test/IBM.WatsonDeveloperCloud.VisualRecognition.UnitTests/VisualRecognitionServiceUnitTests.cs

```diff
@@ -0,0 +1,1811 @@
+        }
+
+        [TestMethod, ExpectedException(typeof(ArgumentNullException))]
+        public void Classify_Post_NoImage_OrName()
+        {
+            IClient client = this.CreateClient();
+            IRequest request = Substitute.For<IRequest>();
+
+            VisualRecognitionService service =
+                new VisualRecognitionService(client);
+
+            var classifications = service.Classify(Arg.Any<byte[]>(), null);
+        }
+
+        [TestMethod, ExpectedException(typeof(ArgumentNullException))]
+        public void Classify_Post_IncorrectOwners()
+        {
+            IClient client = this.CreateClient();
+            IRequest request = Substitute.For<IRequest>();
+
+            VisualRecognitionService service =
+                new VisualRecognitionService(client);
+
+            var classifications = service.Classify(Arg.Any<byte[]>(), Arg.Any<string>(), Arg.Any<string>(), Arg.Any<string[]>(), Arg.Any<string[]>());
+        }
+
+        [TestMethod]
+        public void Classify_Post_Success()
+        {
+            IClient client = this.CreateClient();
+            IRequest request = Substitute.For<IRequest>();
+            client.PostAsync(Arg.Any<string>())
+                  .Returns(request);
+
+            ClassifyPost response = new ClassifyPost()
+            {
+                Images = new List<Classifiers>()
+                {
+                    new Classifiers()
+                    {
+                       _Classifiers = new List<ClassifyPerClassifier>()
+                       {
+                            new ClassifyPerClassifier()
+                            {
+                                Classes = new List<ClassResult>()
+                                {
+                                    new ClassResult()
+                                    {
+                                        _Class = "turtle",
+                                        Score = 0.98f
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                }
+            };
+
+            request.WithHeader(Arg.Any<string>(), Arg.Any<string>())
+                .Returns(request);
+            request.WithArgument(Arg.Any<string>(), Arg.Any<string>())
+                .Returns(request);
+            request.WithArgument(Arg.Any<string>(), Arg.Any<string>())
+                .Returns(request);
+            request.WithBodyContent(Arg.Any<HttpContent>())
+                .Returns(request);
+
+            request.As<ClassifyPost>()
+                .Returns(Task.FromResult(response));
+
+            VisualRecognitionService service =
+                new VisualRecognitionService(client);
+
+            var classifications = service.Classify(new byte[4], "name", "image/jpeg");
+
+            Assert.IsNotNull(classifications);
+            client.Received().PostAsync(Arg.Any<string>());

```

Comment:
@mediumTaj Yes, with the NSubstitute it is necessary to create the mock of all the methods used in the fake class. Validating whether the result as non-null ensures that all methods have been executed. Still, it is important to validate whether the methods Post, Get... were called to ensure that the method we construct is calling the correct verb.